firmware/controllers/core/interpolation.cpp

```diff
@@ -10,10 +10,15 @@
  * @author Dmitry Sidin, (c) 2015
  */
 
+#include "interpolation.h"
+
 #include "global.h"
 
+#include <math.h>
+#include "efilib.h"
 #include "efilib2.h"
-#include "interpolation.h"
+#include "obd_error_codes.h"
```

Comment:
Aren't these included from global.h?

Reply:
They probably are-but global.h causes an explosion of the compiler having to parse things that are never used.  It makes unit testing nearly impossible.  If we can make each component as small and independent as possible, it reduces coupling between parts of the system, which helps make future changes cheaper to implement.

Reply:
Not including it forces you to think carefully about what you're doing, which helps drive up code quality.

Reply:
these are fair arguments, can we have this as an independent PR with includes removed from global.h and added everywhere where they are needed?
So,
1) complete change - not a half change
2) not mixing multiple changes into one mixed change